src/components/date-picker/calendar.tsx

```diff
@@ -240,41 +288,90 @@ export class Calendar extends React.Component<CalendarProps, CalendarState> {
         }
     }
 
-    private toggleMonthView = () => {
-        this.setState({showMonthView: !this.state.showMonthView});
+    private toggleDateSelectView = () => {
+        if (this.state.showYearView && this.state.showMonthView) {
```

Comment:
How about having only `this.state.view` that can be equal to `year|month|default`?